pennylane_lightning/core/src/simulators/lightning_kokkos/catalyst/LightningKokkosSimulator.cpp

```diff
@@ -0,0 +1,575 @@
+// Copyright 2022-2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Kokkos_Complex.hpp>
+#include <Kokkos_Core.hpp>
+
+#include "LightningKokkosSimulator.hpp"
+
+namespace Catalyst::Runtime::Simulator {
+
+auto LightningKokkosSimulator::AllocateQubit() -> QubitIdType {
+    const size_t num_qubits = this->device_sv->getNumQubits();
```

Comment:
Maybe we should take this opportunity to move to `std::size_t`.
```suggestion
    const std::size_t num_qubits = this->device_sv->getNumQubits();
```

Reply:
I will, thanks.